core/query/sql.ml

```diff
@@ -144,201 +137,250 @@ let order_by_clause n =
    returned. This allows these operators to take lists that have any
    element type at all. *)
 
-let pp_comma ppf () = Format.pp_print_string ppf ","
-let pp_comma_separated pp_item = Format.pp_print_list ~pp_sep:pp_comma pp_item
+class printer (quote : string -> string) =
```

Comment:
It seems like there are two strategies for customization being used by the printer class:

1. passing in a (function) argument (e.g. `quote`)
2. extending/overriding the printer class, or creating an object of this class and then customizing it

Why not, e.g., leave the `quote` method abstract and thus force it to be defined by each DB_specific printer, and only use extension strategy (2)?

Reply:
Yes, you're correct. I experimented with both options. I decided upon the approach of having `quote` as a parameter since there's no reasonable default, from what I can see: all 3 DBs do something slightly different.

Upon reflection, I think the best thing to do here would be to make the `quote` method virtual, and then have each printer override it (more specifically, the `default_printer` would still take `quote` as a parameter). This would have the dual advantages of a more uniform design while still not requiring an arbitrary default. Thanks for prompting me to think about this more carefully!